ebuild-writing/functions/src_prepare/eapply/text.xml

```diff
@@ -0,0 +1,123 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<guide self="ebuild-writing/functions/src_prepare/eapply/">
+<chapter>
+<title>Patching with eapply</title>
+<body>
+
+<p>
+The canonical way of applying patches in ebuilds is to use the package
+manager's <c>eapply</c> command, either by calling it explicitly, or by
+assigning the <c>PATCHES</c> variable supported by the default
+<c>src_prepare</c> implementation.
+</p>
+
+<important>
+Applying patches to the sources from the vanilla tarball is <e>strongly</e>
+preferred to distributing a modified tarball.
+</important>
```

Comment:
To be honest, I still find this a little bit confusing, but perhaps because I'm in the position where applying patches is the "obvious" solution. Perhaps make it clear that we're talking about distributing patches vs. creating our own tarball. We don't want to prohibit users from using upstream "patched" tarballs (such as newer kernel sources vs. taking 5.10.0 along with 200+ patches).

While at it, I wonder if we should also point out that for big patches, we prefer hosting them externally and fetching via `SRC_URI` rather than hogging up `FILESDIR`.

Reply:
I've reworded it so that "vanilla tarball" is replaced by "upstream tarball" and "a modified tarball" by "your own modified tarball".